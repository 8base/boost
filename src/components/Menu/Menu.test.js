// @flow
import React from 'react';
import { Menu } from './';

describe('<Menu />', () => {
  it('should shallow Menu', () => {
    const wrapper = shallow(
      <Menu>
        <Menu.Item>Tramman</Menu.Item>
      </Menu>,
    );

    expect(wrapper).toMatchSnapshot();
  });


  it('should shallow Menu item', () => {
    const wrapper = shallow(
      <Menu.Item>Tramman</Menu.Item>,
    );

    expect(wrapper).toMatchSnapshot();
  });


  it('should render Menu components content', () => {
    const wrapper = mount(
      <EightBaseBoostProvider>
        <Menu>
          <Menu.Item>Tramman</Menu.Item>
          <Menu.Item>Gripman</Menu.Item>
          <Menu.Item>Proalliance</Menu.Item>
        </Menu>
      </EightBaseBoostProvider>,
    );

    expect(wrapper.find(Menu.Item).at(0).text()).toBe('Tramman');
    expect(wrapper.find(Menu.Item).at(1).text()).toBe('Gripman');
    expect(wrapper.find(Menu.Item).at(2).text()).toBe('Proalliance');
  });
});

