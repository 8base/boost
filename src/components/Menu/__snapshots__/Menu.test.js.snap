// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should shallow Menu 1`] = `
<Paper
  borderRadius="all"
  padding="none"
>
  <Styled(menu)
    tagName="div"
  >
    <Menu.Item>
      Tramman
    </Menu.Item>
  </Styled(menu)>
</Paper>
`;

exports[`<Menu /> should shallow Menu item 1`] = `
<Styled(menuItem)
  tagName="div"
>
  Tramman
</Styled(menuItem)>
`;
